commit dcaf51b41e259387602774829c45222d0507f90a
Author: Florian Weimer <fweimer@redhat.com>
Date:   Mon Oct 28 14:45:30 2024 +0100

    elf: Change ldconfig auxcache magic number (bug 32231)
    
    In commit c628c2296392ed3bf2cb8d8470668e64fe53389f (elf: Remove
    ldconfig kernel version check), the layout of auxcache entries
    changed because the osversion field was removed from
    struct aux_cache_file_entry.  However, AUX_CACHEMAGIC was not
    changed, so existing files are still used, potentially leading
    to unintended ldconfig behavior.  This commit changes AUX_CACHEMAGIC,
    so that the file is regenerated.
    
    Reported-by: DJ Delorie <dj@redhat.com>
    Reviewed-by: Adhemerval Zanella  <adhemerval.zanella@linaro.org>
    (cherry picked from commit 0a536f6e2f76e3ef581b3fd9af1e5cf4ddc7a5a2)

diff --git a/elf/cache.c b/elf/cache.c
index 8a618e11fa414abb..62d681df42746d55 100644
--- a/elf/cache.c
+++ b/elf/cache.c
@@ -820,7 +820,7 @@ struct aux_cache_entry
   struct aux_cache_entry *next;
 };
 
-#define AUX_CACHEMAGIC		"glibc-ld.so.auxcache-1.0"
+#define AUX_CACHEMAGIC		"glibc-ld.so.auxcache-2.0"
 
 struct aux_cache_file_entry
 {
